core/src/main/java/io/github/lsposed/lspd/service/ActivityManagerService.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * This file is part of LSPosed.
+ *
+ * LSPosed is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * LSPosed is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with LSPosed.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * Copyright (C) 2021 LSPosed Contributors
+ */
+
+package io.github.lsposed.lspd.service;
+
+import android.app.IActivityManager;
+import android.app.IApplicationThread;
+import android.content.IIntentReceiver;
+import android.content.Intent;
+import android.os.Bundle;
+import android.os.IBinder;
+import android.os.RemoteException;
+import android.os.ServiceManager;
+
+public class ActivityManagerService {
+    private static IActivityManager am = null;
+    private static IBinder binder = null;
+
+    public static IActivityManager getActivityManager() {
+        if (binder == null && am == null) {
+            binder = ServiceManager.getService("activity");
+            am = IActivityManager.Stub.asInterface(binder);
+        }
+        return am;
```

Comment:
The binder is cached forever, once the system server restarts, the root service can never get the new binder.

The correct implementation from Sui:
https://github.com/RikkaApps/Sui/blob/master/module/src/main/java/rikka/sui/server/api/SystemServiceBinder.kt